src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/EditAndContinue/EditAndContinueProvider.cs

```diff
@@ -20,160 +20,122 @@ namespace Microsoft.VisualStudio.ProjectSystem.VS.EditAndContinue
-            }
-
-            return HResult.Unexpected;
+            return Invoke(encProvider => encProvider.BuildForEnc(pUpdatePE));
         }
 
         public int ExitBreakStateOnPE()
         {
-            if (_host?.HostSpecificEditAndContinueService is IVsENCRebuildableProjectCfg2 encProvider)
-            {
-                return encProvider.ExitBreakStateOnPE();
-            }
-
-            return HResult.Unexpected;
+            return Invoke(encProvider => encProvider.ExitBreakStateOnPE());
         }
 
         public int StopDebuggingPE()
         {
-            if (_host?.HostSpecificEditAndContinueService is IVsENCRebuildableProjectCfg2 encProvider)
-            {
-                return encProvider.StopDebuggingPE();
-            }
-
-            return HResult.Unexpected;
+            return Invoke(encProvider => encProvider.StopDebuggingPE());
         }
 
         public int GetENCBuildState(ENC_BUILD_STATE[] pENCBuildState)
         {
-            if (_host?.HostSpecificEditAndContinueService is IVsENCRebuildableProjectCfg2 encProvider)
-            {
-                return encProvider.GetENCBuildState(pENCBuildState);
-            }
-
-            return HResult.Unexpected;
+            return Invoke(encProvider => encProvider.GetENCBuildState(pENCBuildState));
         }
 
         public int GetCurrentActiveStatementPosition(uint id, TextSpan[] ptsNewPosition)
         {
-            if (_host?.HostSpecificEditAndContinueService is IVsENCRebuildableProjectCfg2 encProvider)
-            {
-                return encProvider.GetCurrentActiveStatementPosition(id, ptsNewPosition);
-            }
-
-            return HResult.Unexpected;
+            return Invoke(encProvider => encProvider.GetCurrentActiveStatementPosition(id, ptsNewPosition));
         }
 
         public int GetPEidentity(Guid[] pMVID, string[] pbstrPEName)
         {
-            if (_host?.HostSpecificEditAndContinueService is IVsENCRebuildableProjectCfg2 encProvider)
-            {
-                return encProvider.GetPEidentity(pMVID, pbstrPEName);
-            }
-
-            return HResult.Unexpected;
+            return Invoke(encProvider => encProvider.GetPEidentity(pMVID, pbstrPEName));
         }
 
         public int GetExceptionSpanCount(out uint pcExceptionSpan)
         {
-            if (_host?.HostSpecificEditAndContinueService is IVsENCRebuildableProjectCfg2 encProvider)
-            {
-                return encProvider.GetExceptionSpanCount(out pcExceptionSpan);
-            }
+            uint pcExceptionSpanResult = 0;
+            HResult hr = Invoke(encProvider => encProvider.GetExceptionSpanCount(out pcExceptionSpanResult));
+
+            pcExceptionSpan = pcExceptionSpanResult;
 
-            pcExceptionSpan = 0;
-            return HResult.Unexpected;
+            return hr;
         }
 
         public int GetExceptionSpans(uint celt, ENC_EXCEPTION_SPAN[] rgelt, ref uint pceltFetched)
```

Comment:
`pceltFetched` could be `out` rather than `ref`. Any input value is ignored.

Reply:
Just noticed this is probably on an interface and not changeable.

Reply:
Yeah COM interfaces don't have concept of "out".

Reply:
I stand corrected as GetExceptionSpanCount defines an `out` parameter. The IDL states this is `[in, out]`: https://devdiv.visualstudio.com/DevDiv/_git/VS.ref_archive?path=%2Fsrc%2FVSL%2Fidl%2FVBPublic%2Fencbuild.idl&version=GBmaster&line=241&lineStyle=plain&lineEnd=242&lineStartColumn=1&lineEndColumn=1, so either definition is wrong or there's a case where it provides an input (I can't actually find any callers). Either way, I should be respecting the definition.

Reply:
Fixed: https://github.com/dotnet/project-system/pull/4040